src/ring_theory/nilpotent.lean

```diff
@@ -83,6 +83,16 @@ begin
   exact (hx.map f).eq_zero,
 end
 
+lemma ring_hom.ker_is_radical_iff_reduced_of_surjective {R S F} [comm_semiring R] [comm_ring S]
+  [ring_hom_class F R S] {f : F} (hf : function.surjective f) :
+  (ring_hom.ker f).radical ≤ ring_hom.ker f ↔ is_reduced S :=
```

Comment:
Not sure if we should introduce the predicate `ideal.is_radical` now.

Reply:
I've introduced the `ideal.is_radical` predicate along with `is_radical` for a single element.